packages/flame/lib/src/collisions/collision_callbacks.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:meta/meta.dart';
+
+import '../../components.dart';
+
+/// The [CollidableType] is used to determine which other collidables that it
+/// should collide with.
+enum CollidableType {
+  /// Collides with other collidables of type active or passive.
+  active,
+
+  /// Collides with other collidables of type active.
+  passive,
+
+  /// Will not collide with any other collidables.
+  inactive,
+}
+
+/// The [GenericCollisionCallbacks] mixin can be used to get callbacks from the
+/// collision detection system, potentially without using the Flame component
+/// system.
+/// The default implementation used with FCS is [CollisionCallbacks].
+mixin GenericCollisionCallbacks<T> {
+  Set<T>? _activeCollisions;
+
+  /// The objects that the object is currently colliding with.
+  Set<T> get activeCollisions => _activeCollisions ??= {};
+
+  /// Whether the object is currently colliding or not.
+  bool get isColliding {
+    return _activeCollisions != null && _activeCollisions!.isNotEmpty;
+  }
+
+  /// Whether the object is colliding with [other] or not.
+  bool collidingWith(T other) {
+    return _activeCollisions != null && activeCollisions.contains(other);
+  }
+
+  /// [onCollision] is called in every tick when this object is colliding with
+  /// [other].
+  @mustCallSuper
+  void onCollision(Set<Vector2> intersectionPoints, T other) {
```

Comment:
I would call this `onCollisionUpdate`, similar to how we have `onDragStart`, `onDragUpdate` and `onDragEnd`.

Which also makes me wonder if there is such thing as `onCollisionCancel`?

Reply:
`Cancel` comes from Flutter's gesture system, I don't think we have to involve cancel here.